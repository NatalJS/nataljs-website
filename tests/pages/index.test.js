import renderer from 'react-test-renderer';
import Index from 'pages/index';

jest.mock('components/Header', () => 'Header');
jest.mock('components/About', () => 'About');
jest.mock('components/HowTo', () => 'HowTo');
jest.mock('components/Events', () => 'Events');
jest.mock('components/JoinToUs', () => 'JoinToUs');
jest.mock('components/Footer', () => 'Footer');

describe('Page Index', () => {
  let Component;

  beforeAll(() => {
    Component = setup(Index);
  });

  it('should render all page with all components correctly', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
