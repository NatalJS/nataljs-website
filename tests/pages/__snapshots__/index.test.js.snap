// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Index should render all page with all components correctly 1`] = `
Array [
  <Header />,
  <div
    className="container"
  >
    <About />
    <HowTo />
  </div>,
  <section
    className="sign-up"
    id="sign-up"
  >
    <h1
      className="section-title"
    >
      Inscreva-se e participe do evento
    </h1>
    <a
      className="btn btn--white"
      href="https://docs.google.com/forms/d/e/1FAIpQLSdWT3MnHCGjhBp7dn5KYTfd61eysoO93TB8Alz12u2vmXiJ-Q/viewform"
      target="_blank"
    >
      inscrever agora
    </a>
  </section>,
  <section
    className="call-for-papers"
    id="call-for-papers"
  >
    <h1
      className="section-title"
    >
      Call For Papers
    </h1>
    <p>
      Tem uma palestra legal? Tem algo para falar?
    </p>
    <a
      className="btn btn--black"
      href="http://speakerfight.com/events/ii-evento-nataljs"
      targe="_blank"
    >
      Submeta sua palestra
    </a>
  </section>,
  <section
    className="events"
    id="incoming-events"
  >
    <h1
      className="section-title"
    >
      Próximos Eventos?
    </h1>
    <div
      className="events__container"
    >
      <div
        className="events__block"
      >
        <div
          className="col-3 events__time"
        >
          <span>
            15.06
          </span>
          <span>
            8:30H ÀS 12
          </span>
        </div>
        <div
          className="col-9 events__body"
        >
          <div
            className="events__title"
          >
            Em breve serão postados mais informações
          </div>
          <div
            className="events__location"
          >
            IMD UFRN
          </div>
        </div>
      </div>
    </div>
  </section>,
  <JoinToUs />,
  <Footer />,
]
`;
